Microsoft.Azure.CosmosRepository/src/DefaultRepository.cs

```diff
@@ -357,6 +357,37 @@ await container.ReadItemAsync<TItem>(id, partitionKey, cancellationToken: cancel
                 continuationToken);
         }
 
+        /// <inheritdoc/>
+        public async ValueTask<IPage<TItem>> PageAsync(Expression<Func<TItem, bool>> predicate = null, int pageNumber = 1, int pageSize = 25, CancellationToken cancellationToken = default)
+        {
+            Container container = await _containerProvider.GetContainerAsync().ConfigureAwait(false);
+
+            IQueryable<TItem> query = container.GetItemLinqQueryable<TItem>()
+                                               .Where(_repositoryExpressionProvider.Build(predicate));
+
+            Response<int> countResponse = await query.CountAsync(cancellationToken);
+            query = query.Skip(pageSize * (pageNumber - 1))
+                         .Take(pageSize);
+
+            using FeedIterator<TItem> iterator = query.ToFeedIterator();
+
+            FeedResponse<TItem> next = await iterator.ReadNextAsync(cancellationToken)
+                                                     .ConfigureAwait(false);
+
+            List<TItem> results = new();
```

Comment:
This looks good there is some cases where cosmos may not return all the results in a single call to `ReadNextAsync()` so you need to do this in a `while` loop as current paging example does. I'll find the link to the docs shortly

Reply:
See this comment here as part of the initial paging PR, it includes the link I was referring to

https://github.com/IEvangelist/azure-cosmos-dotnet-repository/pull/142#discussion_r763042915